src/main/java/net/dv8tion/jda/internal/interactions/CommandDataImpl.java

```diff
@@ -207,45 +235,106 @@ public CommandDataImpl addSubcommandGroups(@Nonnull SubcommandGroupData... group
         return this;
     }
 
+    @Nonnull
+    @Override
+    public CommandDataImpl setLocalizationMapper(@Nonnull LocalizationMapper localizationMapper) {
+        Checks.notNull(localizationMapper, "Localization mapper");
+
+        this.localizationMapper = localizationMapper;
+        return this;
+    }
+
     @Nonnull
     @Override
     public CommandDataImpl setName(@Nonnull String name)
     {
-        Checks.inRange(name, 1, 32, "Name");
-        if (type == Command.Type.SLASH)
-        {
-            Checks.matches(name, Checks.ALPHANUMERIC_WITH_DASH, "Name");
-            Checks.isLowercase(name, "Name");
-        }
+        checkName(name);
         this.name = name;
         return this;
     }
 
+    @Nonnull
+    @Override
+    public CommandDataImpl setNameLocalization(@Nonnull DiscordLocale locale, @Nonnull String name)
+    {
+        //Checks are done in LocalizationMap
+        nameLocalizations.setTranslation(locale, name);
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public SlashCommandData setNameLocalizations(@Nonnull Map<DiscordLocale, String> map)
+    {
+        nameLocalizations.setTranslations(map);
+        return this;
+    }
+
+    public CommandDataImpl setNameLocalizations(@Nonnull LocalizationMap nameLocalizations)
+    {
+        this.nameLocalizations = nameLocalizations;
+        return this;
+    }
```

Comment:
Is there a reason we support this instead of going through the `setNameLocalizations(Map)` or looping for the `setNameLocalization(DiscordLocale, String)`?